src/frontend/parallelization_phase.nit

```diff
@@ -39,10 +39,9 @@ private class ParallelizationPhase
 			toolcontext.error(nmethdef.location, "Syntax error: only a method can be threaded.")
 			return
 		end
-		if nmethdef.n_signature.n_params.length != 0 then
-			toolcontext.error(nmethdef.location, "Syntax error: parametrized method not supported yet.")
-			return
-		end
+
+		#TODO: check for self calls
```

Comment:
If you mark the mmethod as `is_top_level` then use of self will be forbidden.
The other way is to allow self, but document that it refer the the thread object, but this is not POLA :(